Sources/XcodeGenKit/PBXProjGenerator.swift

```diff
@@ -799,8 +798,12 @@ public class PBXProjGenerator {
                 }
             // Embedding handled by iterating over `carthageDependencies` below
             case .package(let product):
-                guard let packageReference = packageReferences[dependency.reference] else {
-                    return
```

Comment:
I found that this code would have an issue when there is no package reference. This early return will cause the outputting unfinished project file. Currently, This does not occur because current project validation works well, but I think it would be better to make this as `continue` ASAP. => https://github.com/yonaskolb/XcodeGen/pull/796/commits/dd7737354dd00baac77bd19f75b16ff834f11c00